src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceRange.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+
+namespace Microsoft.FeatureManagement.FeatureFilters
+{
+    /// <summary>
+    /// The recurrence range describes a date range over which the time window repeats.
+    /// </summary>
+    public class RecurrenceRange
+    {
+        /// <summary>
+        /// The recurrence range type.
+        /// </summary>
+        public RecurrenceRangeType Type { get; set; }
+
+        /// <summary>
+        /// The date to stop applying the recurrence pattern.
+        /// </summary>
+        public DateTimeOffset? EndDate { get; set; }
+
+        /// <summary>
+        /// The number of times to repeat the time window.
+        /// </summary>
+        public int NumberOfOccurrences { get; set; } = 1;
```

Comment:
Yes. It makes more sense to let this property be nullable. Decided to remove the default value and change the type to int?

Reply:
After reconsidering it, I still prefer to set a default value here. I agree that it would be more intuitive for this property to be nullable since it is only required when the recurrence range type is "Numbered". But I want to argue for the current choice:

The main reason I choose to set a default value and avoid using type nullable int is:
If the type is nullable int, then if we want to refer it, we will need to write the code like this: 
``` C#
if(Range.NumberOfOccurrences.Value < 1)
```
And also we need to add null check for `Range.NumberOfOccurrences`. I want to simplify the code.


The inconsistency is that "EndDate" property is nullable. 

We do have two nullable properties in RecurrencePattern which are "DayOfMonth" and "Month". The reason why these two properties have no default value is that I cannot come up with a reasonable default value for them. If everyone agree that the DayOfMonth and Month should be 1 by default. I will definitly set a default value for them. Because in this way, we can simplify our validation process by removing the null check for them.

In fact, I just found I forgot to cover the case if the "EndDate" is null when RecurrenceRangeType is "EndDate" in unit tests.

So my current thought is to keep the default value for the "NumberOfOccurrences" and set a default value DateTimeOffset.MaxValue for "EndDate". Do you like this idea?

Reply:
If EndDate is `MaxValue` by default, why wouldn't `NumberOfOccurrences` be `int.MaxValue` ?